### ISSUE TITLE
Lost content 

### ISSUE THREAD

**Issue by Qeeadde** [typora/typora-issues#4291]
I ran into a problem that the content of file access is empty when the computer suddenly crashed and the software did not exit properly. The file is actually 124KB as normal, while the file recovered from the draft file is 120KB. When opening the empty file with vscode, it shows the binary file cannot be opened, while normal files can be opened in markdown format. In other words, because of the non-normal exit of the software, the content could not be recognized as a normal markdown file, and it shows empty content...

my version is 0.9.98(beta) Typora for windows/liunx

Thanks

**Comment by gnzlbg** [comment]
I just lost a big file with 0.9.98 (beta) for windows/linux on linux.

I rebooted the computer as usual, typora exited, and on restart, the file was empty.

**Comment by abnerlee** [comment]
@gnzlbg @Qeeadde There there "edited" or "unsaved" mark (the * mark) on Typora window before reboot? Or there is unsaved content before reboot?

---

Does https://support.typora.io/Version-Control/ help?

**Comment by Qeeadde** [comment]
In my case, there were unsaved file contents before the computer crashed. Some of my files were not saved. Files have been edited for a few days. I also regularly press the save shortcut key. After restarting the computer and launching the applications, the original workspace and files are usually restored. And this time it didn't restore, and when I went back to the file and opened it again, the contents of the file were just empty. In addition, what I was thinking before was wrong. Later, I tried to force a reboot without saving and it didn't reproduce the problem I encountered. I think it's harder to reproduce, so it's a special case that occurs with a small probability. I don't really understand it either. I didn't use the auto-save feature, and I was able to recover most of the content using the recovery draft, which worked relatively well.
